cli/external_workspaces.go

```diff
@@ -0,0 +1,326 @@
+		}
+
+		organization, err := orgContext.Selected(inv, client)
+		if err != nil {
+			return xerrors.Errorf("get current organization: %w", err)
+		}
+
+		template, err := client.TemplateByName(inv.Context(), organization.ID, templateName.Value.String())
+		if err != nil {
+			return xerrors.Errorf("get template by name: %w", err)
+		}
+
+		var resources []codersdk.WorkspaceResource
+		var templateVersionID uuid.UUID
+		if templateVersion == nil || templateVersion.Value.String() == "" {
+			templateVersionID = template.ActiveVersionID
+		} else {
+			version, err := client.TemplateVersionByName(inv.Context(), template.ID, templateVersion.Value.String())
+			if err != nil {
+				return xerrors.Errorf("get template version by name: %w", err)
+			}
+			templateVersionID = version.ID
+		}
+
+		resources, err = client.TemplateVersionResources(inv.Context(), templateVersionID)
+		if err != nil {
+			return xerrors.Errorf("get template version resources: %w", err)
+		}
+		if len(resources) == 0 {
+			return xerrors.Errorf("no resources found for template version %q", templateVersion.Value.String())
+		}
+
+		var hasExternalAgent bool
+		for _, resource := range resources {
+			if resource.Type == "coder_external_agent" {
+				hasExternalAgent = true
+				break
+			}
+		}
+
+		if !hasExternalAgent {
+			return xerrors.Errorf("template version %q does not have an external agent. Only templates with external agents can be used for external workspace creation", templateVersion.Value.String())
+		}
+
+		err = createHandler(inv)
+		if err != nil {
+			return err
+		}
+
+		workspace, err := client.WorkspaceByOwnerAndName(inv.Context(), codersdk.Me, workspaceName, codersdk.WorkspaceOptions{})
+		if err != nil {
+			return xerrors.Errorf("get workspace by name: %w", err)
+		}
+
+		externalAgents, err := fetchExternalAgents(inv, client, workspace, workspace.LatestBuild.Resources)
+		if err != nil {
+			return xerrors.Errorf("fetch external agents: %w", err)
+		}
+
+		return printExternalAgents(inv, workspace.Name, externalAgents)
+	}
+	return cmd
+}
+
+// externalWorkspaceAgentInstructions prints the instructions for an external agent.
+func (r *RootCmd) externalWorkspaceAgentInstructions() *serpent.Command {
+	client := new(codersdk.Client)
+	formatter := cliui.NewOutputFormatter(
+		cliui.ChangeFormatterData(cliui.TextFormat(), func(data any) (any, error) {
+			agent, ok := data.(externalAgent)
+			if !ok {
+				return "", xerrors.Errorf("expected externalAgent, got %T", data)
+			}
+
+			var output strings.Builder
+			_, _ = output.WriteString(fmt.Sprintf("Please run the following commands to attach agent %s:\n", cliui.Keyword(agent.AgentName)))
+			_, _ = output.WriteString(fmt.Sprintf("%s\n", pretty.Sprint(cliui.DefaultStyles.Code, fmt.Sprintf("export CODER_AGENT_TOKEN=%s", agent.AuthToken))))
+			_, _ = output.WriteString(pretty.Sprint(cliui.DefaultStyles.Code, fmt.Sprintf("curl -fsSL %s | sh", agent.InitScript)))
+
+			return output.String(), nil
+		}),
```

Comment:
This chunk of code seems to be used multiple times, can you not reuse `printExternalAgents()`?